docs/en/advanced/Security.md

```diff
@@ -219,54 +219,77 @@ ACLs can be managed by two ways:
 
 To enable a particular user to impersonate other groups, set the
 `alluxio.master.security.impersonation.<USERNAME>.groups` property, where again `<USERNAME>` is
-the name of the connection user.
+the name of the Alluxio client user.
 
 Similar to the above, the value is a comma-separated list of groups and the wildcard value `*`
 can be used to indicate all groups.
 Some examples:
 
 - `alluxio.master.security.impersonation.alluxio_user.groups=group1,group2`
-means the connection user `alluxio_user` is allowed to impersonate any users from groups `group1` and `group2`.
+means the Alluxio client user `alluxio_user` is allowed to impersonate any users from groups `group1` and `group2`.
 - `alluxio.master.security.impersonation.client.groups=*`
-means the connection user `client` is allowed to impersonate users from any group.
+means the Alluxio client user `client` is allowed to impersonate users from any group.
 
-In summary, impersonation of a given user is only enabled if at least one of the two impersonation
-properties are be set; setting both are allowed for the same user.
+In summary, to enable an Alluxio client to impersonate other users, at least one of the two
+impersonation properties must be set on servers; setting both are allowed for the same Alluxio
+client user.
 
 ### Client Configuration
 
-After enabling impersonation on the master for a given user,
+After enabling impersonation on the master for a given Alluxio client user,
 the client must indicate which user it wants to impersonate.
 This is configured by the `alluxio.security.login.impersonation.username` property.
 
-If the property is set to an empty string or `_NONE_`, impersonation is disabled.
-If the property is set to `_HDFS_USER_`, the Alluxio client will impersonate as the same user as
-the HDFS client when using the Hadoop compatible client.
-This property can also be set to a particular string to indicate a user name.
+If the property is set to an empty string or `_NONE_`, impersonation is disabled, and the Alluxio
+client will interact with Alluxio servers as the Alluxio client user.
+If the property is set to `_HDFS_USER_`, the Alluxio client will connect to Alluxio servers as the
+Alluxio client user, but impersonate as the Hadoop client user when using the Hadoop compatible
+client.
+
+### Exceptions
+
+The most common impersonation error applications may see is something like
+`User yarn is not configured for any impersonation. impersonationUser: foo`. This is most likely
+due to the fact that the Alluxio servers have not been configured to enable impersonation for that
+user. To fix this, the Alluxio servers must be configured to enable impersonation for the user in question (yarn in the example error message).
+
+Please read this [blog post](http://www.alluxio.com/blog/alluxio-developer-tip-why-am-i-seeing-the-error-user-yarn-is-not-configured-for-any-impersonation-impersonationuser-foo) for more tips.
```

Comment:
Do we have a similar doc hosted on OS site?

Reply:
Not that I know of...